### ISSUE TITLE
azcopy generate SIGTTIN when in background

### ISSUE THREAD

**Issue by bakoontz2** [Azure/azure-storage-azcopy#2038]
##### Note: The version is visible when running AzCopy without any argument
10.15.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Linux

##### Note: Please remove the SAS to avoid exposing your credentials. If you cannot remember the exact command, please retrieve it from the beginning of the log file.
Minimal test case:
```
/bin/sh
azcopy cp "SOURCE" "DEST" --recursive  --include-after=2022-12-18
```

### What problem was encountered?
If the script above is run as a background process, a SIGTTIN is generated by azcopy, causing the process to hang:

[1]+  Stopped(SIGTTIN)        ./azcopy_test.sh

Running in the foreground does not stop the process for input, so the signal is being falsely generated by azcopy.

### How can we reproduce the problem in the simplest way?

1.  Create a script called azcopy_test.sh with the code above.
2.  Run in background: ./azcopy_test.sh &
3.  Observer the generation of the SIGTTIN signal.
4.  Run in foreground, no signal generated.

### Have you found a mitigation/solution?
A temporary workaround is to trap and ignore the SIGTTIN:

trap "" TTIN

But this does not mitigate the buggy behavior itself.

**Comment by bakoontz2** [comment]
I believe I found the triggering issue:  There was a message buried deep in the logs about a certificate failure.  Installing ca-certificates package resolved this.

However, I still believe the signal being generated in response to this issue is itself in error and misleading.  azcopy should exit gracefully, not simply hang for an indeterminate period of time.

**Comment by siminsavani-msft** [comment]
Hi @bakoontz2 ! Thank you for updating us on what the issue was. I will relay this back to the team to make a note of this so we can improve the error logging.

**Comment by adreed-msft** [comment]
Does this behavior occur when AzCopy is not running in the background? `SIGTTIN` is usually sent to a background process to tell it that it will not be able to receive data from `stdin`. This isn't actually generated by AzCopy itself, but is sent to AzCopy.

The default behavior for `SIGTTIN` is to actually outright stop the process.

**Comment by bakoontz2** [comment]
When run in the foreground, azcopy simply hangs:

```
2023/02/13 19:25:18 ==> REQUEST/RESPONSE (Try=9/2.254433223s, OpTime=4m1.8914789
83s) -- REQUEST ERROR
   GET https://somepath?restype=directory&se=2023-03-26T00%3A18%3A55Z&sig=-REDACTED-&sp
=rwdlc&spr=https&srt=sco&ss=f&st=2022-03-25T16%3A18%3A55Z&sv=2020-08-04&timeout=
901
   User-Agent: [AzCopy/10.15.0 Azure-Storage/0.8.0 (go1.17.9; linux)]
   X-Ms-Client-Request-Id: [2c17d7c1-33f8-41af-4805-7a5f6aa89327]
   X-Ms-Version: [2020-10-02]
   -----------------------------------------------------------------------------
---
   ERROR:
-> github.com/Azure/azure-pipeline-go/pipeline.NewError, /home/vsts/go/pkg/mod/g
ithub.com/!azure/azure-pipeline-go@v0.2.4-0.20220425205405-09e6f201e1e4/pipeline
/error.go:157
HTTP request failed

Get "https://somepath?restype=directory&se=2023-03-26T00%3A18%3A55Z&sig=-REDACTED-&sp=r
wdlc&spr=https&srt=sco&ss=f&st=2022-03-25T16%3A18%3A55Z&sv=2020-08-04&timeout=90
1": x509: certificate signed by unknown authority
```

**Comment by adreed-msft** [comment]
> When run in the foreground, azcopy simply hangs:
> 
> ```
> 2023/02/13 19:25:18 ==> REQUEST/RESPONSE (Try=9/2.254433223s, OpTime=4m1.8914789
> 83s) -- REQUEST ERROR
>    GET https://somepath?restype=directory&se=2023-03-26T00%3A18%3A55Z&sig=-REDACTED-&sp
> =rwdlc&spr=https&srt=sco&ss=f&st=2022-03-25T16%3A18%3A55Z&sv=2020-08-04&timeout=
> 901
>    User-Agent: [AzCopy/10.15.0 Azure-Storage/0.8.0 (go1.17.9; linux)]
>    X-Ms-Client-Request-Id: [2c17d7c1-33f8-41af-4805-7a5f6aa89327]
>    X-Ms-Version: [2020-10-02]
>    -----------------------------------------------------------------------------
> ---
>    ERROR:
> -> github.com/Azure/azure-pipeline-go/pipeline.NewError, /home/vsts/go/pkg/mod/g
> ithub.com/!azure/azure-pipeline-go@v0.2.4-0.20220425205405-09e6f201e1e4/pipeline
> /error.go:157
> HTTP request failed
> 
> Get "https://somepath?restype=directory&se=2023-03-26T00%3A18%3A55Z&sig=-REDACTED-&sp=r
> wdlc&spr=https&srt=sco&ss=f&st=2022-03-25T16%3A18%3A55Z&sv=2020-08-04&timeout=90
> 1": x509: certificate signed by unknown authority
> ```

Makes sense... We shouldn't be retrying on cert errors though. SIGTTIN is expected behavior on behalf of Linux, the retry is the real issue here

**Comment by bakoontz2** [comment]
Interesting about SIGTTIN and thanks for the followup!
